cli/src/commands/create/dataset.rs

```diff
@@ -1,29 +1,15 @@
 use anyhow::{Context, Result};
 use log::info;
-use once_cell::sync::Lazy;
-use regex::Regex;
 use reinfer_client::{
     resources::dataset::EntityDefs, Client, DatasetFullName, NewDataset, SourceIdentifier,
 };
 use structopt::StructOpt;
 
-static RX_FULL_NAME: Lazy<Regex> =
-    Lazy::new(|| Regex::new("^[A-Za-z0-9-_.]{1,1024}/[A-Za-z0-9-_]{1,256}").unwrap());
-
-/// Ensures the name provided conforms to the <owner>/<name> structure
-fn validate_dataset_name(s: String) -> Result<(), String> {
-    if !RX_FULL_NAME.is_match(&s) {
-        return Err(s);
-    }
-
-    Ok(())
-}
-
```

Comment:
as far as I can tell we're removing this and the only validation left is that it's two parts split by `/`, when using `FullName::from_str`. Do we not want to keep the regex validation?

Reply:
No, I don't think we do. We don't validate any other ids with regexes in this way. We leave pretty much all the validation up to the API and report those errors back to the user, which I think is actively good - there's no point reimplementing it all client side.

This change brings it into line with how we talk about source/bucket ids.